src/components/simple-color-picker/simple-color-picker.component.tsx

```diff
@@ -41,21 +68,53 @@ const SimpleColorPicker = (props) => {
     ...rest
   } = props;
 
-  const filteredChildren = useMemo(() => React.Children.toArray(children), [
-    children,
-  ]);
+  const filteredChildren = useMemo(
+    () =>
+      React.Children.toArray(children).filter((child) =>
+        React.isValidElement(child)
+      ) as React.FunctionComponentElement<
+        SimpleColorProps & { ref: React.RefObject<HTMLInputElement> }
+      >[],
+    [children]
+  );
+
+  const hasProperChildren = useMemo(() => {
+    let hasSimpleColorChildren = true;
+
+    React.Children.toArray(children).forEach((child) => {
+      if (
+        typeof child === "string" ||
+        (React.isValidElement(child) &&
+          (child.type as React.FunctionComponent).displayName !== "SimpleColor")
```

Comment:
I think that we need a safeguard in case a child is not a valid element

Reply:
I'm not sure if I understand to be honest

Reply:
if one of the values passed down in an array will be undefined or a boolean it will break the component

Reply:
ok, I forgot that the falsy values would be filtered out in `toArray`